app/src/main/java/com/github/sdpteam15/polyevents/model/database/local/adapter/LocalCacheAdapter.kt

```diff
@@ -221,179 +215,142 @@ class LocalCacheAdapter(private val db: DatabaseInterface) : DatabaseInterface {
-                                    key,
-                                    collection.value
-                                )
-                            )
-                    }
-                }
-
-                //If from getEntity
-                if (element != null && id != null) {
-                    if (element.value != it.value[id]) {
-                        element.postValue(it.value[id], it.sender)
-                        if (element.value != null)
-                            ended.postValue(true, it.sender)
-                    }
-                    if (element.value == null)
-                        ended.postValue(false, it.sender)
-                }
-
-                //If from getMapEntity
-                if (elements != null) {
-                    ids.apply({ ids ->
-                        val map = mutableMapOf<String, T>()
-                        ids.forEach { eid ->
-                            it.value[eid].apply { value ->
-                                if (elements[eid] != value)
-                                    map[eid] = value
+        PolyEventsApplication.application.applicationScope.launch(Dispatchers.IO) {
+            val date = PolyEventsApplication.application.localDatabase.genericEntityDao()
+                .lastUpdateDate(collection.value)
+            val temp = ObservableMap<String, T>()
+            db.getMapEntity(
+                temp,
+                null,
+                date.apply { date -> { it.whereGreaterThan(LogAdapter.LAST_UPDATE, date) } },
+                collection,
+                adapter
+            ).observeOnce {
+                if (it.value) {
+                        try
+                        {
+                            PolyEventsApplication.application.applicationScope.launch(Dispatchers.IO) {
+                                temp.forEach { element ->
+                                    PolyEventsApplication.application.localDatabase.genericEntityDao()
+                                        .insert(
+                                            LocalAdapter.toDocument(
+                                                (collection.adapter as AdapterToDocumentInterface<T>).toDocumentWithoutNull(
+                                                    element.value
+                                                ),
+                                                element.key,
+                                                collection.value
+                                            )
+                                        )
+                                }
+                                ended.postValue(true, it.sender)
                             }
                         }
-                        if (map.isNotEmpty()) {
-                            elements.putAll(map, it.sender)
+                        catch (e : Exception){
                             ended.postValue(true, it.sender)
```

Comment:
Shouldn't we post false instead of true if there is an exception

Reply:
no is't if we don't have the proper adapter